sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -45,35 +45,28 @@ case class CatalogFunction(
  */
 case class CatalogStorageFormat(
     locationUri: Option[String],
-    inputFormat: Option[String],
-    outputFormat: Option[String],
-    serde: Option[String],
-    compressed: Boolean,
-    serdeProperties: Map[String, String]) {
+    provider: Option[String],
```

Comment:
it'd be good to document in classdoc what these mean, and in what cases can it be None, and the behavior for hive.